### ISSUE TITLE
BUG: Import from Strava fails with pydantic_core.ValidationError

### ISSUE THREAD

**Issue by fortysix2ahead** [stravalib/stravalib#598]
### Stravalib version checks

- [X] I have tested this in a new clean environment with only stravalib and core python files.

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/stravalib/) of stravalib.

- [ ] I have confirmed this bug exists on the [main branch](https://github.com/stravalib/stravalib) of stravalib.

### What operating system are you seeing the problem on?

Mac

### What version of python or you running?

```python
3.10.15
```

### Reproducible Example

Sorry for not creating a small example. The code which imports from Strava is available here: https://github.com/fortysix2ahead/tracs/blob/e2c98641923113272e104e41f4f8ee41196fad85/tracs/plugins/strava.py#L59
and I'm following this bug report here: https://github.com/fortysix2ahead/tracs/issues/370

### Issue Description

There seems to be a bug which prevents Stravalib from reading JSONs for DetailedActivity. Stravalib seems to expect that all times are timezone-aware, however Strava delivers a mix or timezone-aware and naive times. This result in the following error:

```
pydantic_core._pydantic_core.ValidationError: 2 validation errors for DetailedActivity
start_date_local
  Input should have timezone info [type=timezone_aware, input_value='2024-11-04T15:28:42', input_type=str]
    For further information visit https://errors.pydantic.dev/2.9/v/timezone_aware
laps.0.start_date_local
  Input should have timezone info [type=timezone_aware, input_value='2024-11-04T15:28:42', input_type=str]
    For further information visit https://errors.pydantic.dev/2.9/v/timezone_aware
```

The payload Strava sends looks like this:

```json
{
  ...
  "start_date" : "2024-11-04T14:28:42Z",
  "start_date_local" : "2024-11-04T15:28:42",
  "timezone" : "(GMT+01:00) Europe/Berlin",
  ...
}
```

This happens to the start_date of an activity itself, as well as for laps. Maybe the Pydantic annotations in DetailedActvity need to be made a bit more lazier.

### Expected Behavior

Import of Strava-provided JSONs shall work.

### Your environment

aiohappyeyeballs==2.4.0
aiohttp==3.10.5
aiohttp-jinja2==1.6
aiosignal==1.3.1
annotated-types==0.7.0
appdirs==1.4.4
arrow==1.3.0
async-timeout==4.0.3
attrs==24.2.0
babel==2.16.0
beautifulsoup4==4.12.3
bottle==0.13.2
bumpver==2023.1129
cattrs==24.1.2
certifi==2024.2.2
chardet==5.2.0
charset-normalizer==3.3.2
click==8.1.7
click-shell==2.1
colorama==0.4.6
confuse==2.0.1
DateTimeRange==2.3.0
decorator==5.1.1
Deprecated==1.2.14
docutils==0.21.2
dynaconf==3.2.6
exceptiongroup==1.2.0
flit==3.9.0
flit_core==3.9.0
frozenlist==1.4.1
fs==2.4.16
geojson==3.1.0
ghp-import==2.1.0
gpxpy==1.6.2
haversine==2.8.1
idna==3.6
iniconfig==2.0.0
isodate==0.7.2
Jinja2==3.1.4
lexid==2021.1006
linkify-it-py==2.0.3
looseversion==1.3.0
lxml==5.3.0
Markdown==3.5.2
markdown-it-py==3.0.0
MarkupSafe==2.1.5
marshmallow==3.21.3
mbstrdecoder==1.1.3
mdit-py-plugins==0.4.0
mdurl==0.1.2
mergedeep==1.3.4
mkdocs==1.6.1
mkdocs-click==0.8.1
mkdocs-get-deps==0.2.0
mkdocs-material==9.5.43
mkdocs-material-extensions==1.3.1
more-itertools==10.5.0
msgpack==1.0.7
multidict==6.0.5
oauthlib==3.2.2
orjson==3.10.11
packaging==23.2
paginate==0.5.6
pathspec==0.12.1
Pint==0.23
platformdirs==4.3.6
pluggy==1.5.0
ply==3.11
pydantic==2.9.2
pydantic_core==2.23.4
Pygments==2.17.2
pymdown-extensions==10.7
pytest==8.3.3
python-dateutil==2.9.0.post0
pytz==2024.1
PyYAML==6.0.2
pyyaml_env_tag==0.1
regex==2023.12.25
requests==2.32.3
requests-cache==1.2.1
requests-oauthlib==2.0.0
rich==13.9.4
rule-engine==4.5.0
six==1.16.0
soupsieve==2.5
stravalib==2.0
tcxreader==0.4.10
textual==0.79.1
textual-dev==1.6.1
textual-serve==1.1.1
toml==0.10.2
tomli==2.0.1
tomli_w==1.0.0
-e git+https://github.com/fortysix2ahead/tracs.git@e2c98641923113272e104e41f4f8ee41196fad85#egg=tracs
typepy==1.3.2
types-python-dateutil==2.8.19.20240106
typing_extensions==4.9.0
tzlocal==5.2
uc-micro-py==1.0.3
url-normalize==1.4.3
urllib3==2.2.0
watchdog==4.0.0
wrapt==1.16.0
yarl==1.9.4

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by lwasser** [comment]
Hey there, @fortysix2ahead. Thank you for opening this issue. Can you please help us isolate the issue by providing a small code snippet that we can use to reproduce the problem? It's challenging for us as volunteers to have to dig through another code base to understand the issue but we do want to help if we can!  An isolated example that we can reproduce will be super helpful!

**Comment by fortysix2ahead** [comment]
Hi @lwasser I created a test case here:
https://github.com/fortysix2ahead/tracs/blob/fix/370-strava-import-fails-with-pydantic-error/test/test_strava_bug_370.py

I think the culprit is this line, where the type is set to AwareDatetime:
https://github.com/stravalib/stravalib/blob/ef78fc6d96b09409d544b99ae5c895906e4e4fee/src/stravalib/strava_model.py#L631

I checked on the main branch, there the type is datetime (which is probably correct). Maybe it's already been fixed?

**Comment by lwasser** [comment]
hey @fortysix2ahead YES this was a change recently made to the API. strava_model.py is a model generated by Strava's swagger.json. Before i try to debug further, can you test whether this fixes things for you? If it does I will likely add an additional test to our test suite (similar to the one you linked to above). If it doesn't, then I can try to troubleshoot!  

 If I don't have to troubleshoot, then I will work on another open issue! 
Thank you!

**Comment by lwasser** [comment]
Ok i was curious. i just tested it out and WAS able to recreate the issue on stravalib 2.0 and it is not there on stravalib 2.1 after the model update. 

note:
I used this rather than `json.loads` . 
`DetailedActivity.model_validate_json(json_str)`

**Comment by fortysix2ahead** [comment]
@lwasser Thanks a lot, so we can close this issue. I have to wait for stravalib 2.1, until then I'll implement a small workaround and remove it once 2.1 is released.

**Comment by lwasser** [comment]
Of course! I released 2.1 yesterday to pypi! So you should be good to go!

**Comment by fortysix2ahead** [comment]
@lwasser Yes, I can confirm that stravalib 2.1.0 fixes the issue. I'll keep the test case for future reference and merged the PR in Tracs (https://github.com/fortysix2ahead/tracs/pull/389).

Only one thing: you were right, `DetailedActivity.model_validate_json(json_str)` works, while `DetailedActivity.model_validate_json( loads( json_str) )` does not. This is because `start_date` is of type str (the isoformat of datetime), while a datetime object is expected. That's why the latter fails.
